framework/Optimizers/parentSelectors/parentSelectors.py

```diff
@@ -86,29 +86,48 @@ def tournamentSelection(population,**kwargs):
   fitness = kwargs['fitness']
   nParents= kwargs['nParents']
   pop = population
-
   popSize = population.values.shape[0]
 
+  if 'rank' in kwargs:
+    rank = kwargs['rank']
```

Comment:
Where do you assign the "rank"? Through the XML input file? If so, please update the user manual. 
I have checked the new added test, and I do not see the "rank" option, do you set it internally? If so, where do you compute the "rank"?

Reply:
The rank is used only in MO optimization. It is generated within the _useRealization in the geneticAlgorithm.py
Since we have available only single objective optimization, we are not testing lines 119-130. We are splitting the development into multiple PRs. Once the crowding distance is merged (#1584), we will create a PR with full MO optimization which includes a new test for lines 119-130.

Reply:
I see. In this case, I think it is better to put some comments regarding these lines for MO. Since you are actively working on this, it is fine to me.

Reply:
Comments added